storage/remote/queue_manager.go

```diff
@@ -927,6 +925,7 @@ func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.Ti
 		if backoff > s.qm.cfg.MaxBackoff {
 			backoff = s.qm.cfg.MaxBackoff
 		}
+		span.Finish()
```

Comment:
What about the `return` cases?

Reply:
Yup, incoming!

Reply:
This works now, but I don't love that we have to remember to put the `span.Finish()` calls in so many places. Would it be worth breaking the actual try into a function, possibly even an anonymous function then we could use defer?